engine/src/main/java/io/camunda/zeebe/engine/processing/job/ScheduledTrigger.java

```diff
@@ -0,0 +1,62 @@
+/*
+ * Copyright Camunda Services GmbH and/or licensed to Camunda Services GmbH under
+ * one or more contributor license agreements. See the NOTICE file distributed
+ * with this work for additional information regarding copyright ownership.
+ * Licensed under the Zeebe Community License 1.1. You may not use this file
+ * except in compliance with the Zeebe Community License 1.1.
+ */
+package io.camunda.zeebe.engine.processing.job;
+
+import io.camunda.zeebe.engine.processing.streamprocessor.ReadonlyProcessingContext;
+import io.camunda.zeebe.engine.processing.streamprocessor.StreamProcessorLifecycleAware;
+import io.camunda.zeebe.util.sched.ScheduledTimer;
+import java.time.Duration;
+
+public abstract class ScheduledTrigger implements StreamProcessorLifecycleAware {
```

Comment:
@saig0 I have a question about future implementation: should we use side effects in the processor or it is okay to call a new checker directly in accepting?

Reply:
I think that using side effects fits more because in another way we are scheduling execution potentially before we have an instance for the execution, but maybe I'm wrong, WDYT?

Reply:
Yes. At the moment, we should call the checker as side effects. Otherwise, the scheduler could read the entry before it is persisted. However, this would only happen for very short periods.